src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final int DEFAULT_SIZE = 0;
+    private Object[] elements;
+    private int capacity;
+    private int size;
+
+    public ArrayList() {
+        elements = new Object[DEFAULT_CAPACITY];
+        size = DEFAULT_SIZE;
+        capacity = DEFAULT_CAPACITY;
+    }
+
+    private void resize() {
+        capacity = (int) (capacity * 1.5);
+        Object[] elementsResized = new Object[capacity];
+        System.arraycopy(elements, 0, elementsResized, 0, elements.length);
+        elements = elementsResized;
+    }
+
+    private void checkIfIndexMoreOrEqualSize(int index) {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("The index " + index + " is unavailable");
+        }
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == capacity) {
+            resize();
+        }
+        elements[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("The index " + index + " is unavailable");
+        }
```

Comment:
The index validation logic is repeated here. Consider refactoring this logic into a separate private method to adhere to the DRY principle. You already have a method `checkIfIndexMoreOrEqualSize` that checks index bounds; you might want to create a similar method for this case or adjust the existing one to handle both scenarios.